### PR TITLE
 Incorporating argparse for command-line argument handling

### PR DESCRIPTION
**Pull request by meriam04**
**Description**:
This Pull Request implements the use of `argparse` module for improved command-line argument handling in the project. `argparse` provides a convenient way to parse command-line arguments, making the project more user-friendly and robust.

**Changes Made**:
- Integrated `argparse` module to handle command-line arguments effectively.
- Added command-line options for various functionalities:
  - `-b` or `--binary-face` for enabling binary face mode.
  - `-f` or `--get-frames` for specifying whether to get frames.
  - `-c` or `--crop-images` for indicating whether to crop images.
  - `-l` or `--log` for enabling logging.
  - `-m` or `--plot-matlab` for plotting with Matlab.
  - `-r` or `--plot-result` for plotting results.
- Updated relevant functions to accept command-line arguments using `argparse`.
- If the flag is not set, the associated variable is `False`. This was the most intuitive solution (rather than integrating default values), and is not too complex given the shorthand versions of the flags.

**Testing Done**:
- Manually tested the script with various combinations of command-line arguments to ensure proper functionality and default values handling.

### CODE REVIEW

File: data_processing/data_processing/face/process_data.py [hunk 1]
```diff
@@ -224,10 +225,35 @@ def process_data(
...
+    parser.add_argument(
+        "-f", "--get-frames", action="store_true", help="Whether to get frames."
+    )
+    parser.add_argument(
+        "-c", "--crop-images", action="store_true", help="Whether to crop images."
+    )
+    parser.add_argument(
+        "-l", "--log", action="store_true", help="Whether to log debug messages."
+    )
+    args = parser.parse_args()
```
Comment (phonycrow) [inline data_processing/data_processing/face/process_data.py:249]: Can you put the arg parsing in a function that returns the args object?

**Review by phonycrow — COMMENTED** [review]
The arg parsing looks good. I just think that it should be put in its own function, and then this PR can be merged.

File: data_processing/data_processing/face/process_data.py (hunk 1 above)
Comment (meriam04) [reply data_processing/data_processing/face/process_data.py]: done!

**Review by phonycrow — APPROVED** [review]
Everything looks good. Great work!

**Pull request merged**